Fifth task commit. Waiting for review.

Radu-Stroe opened the pull request:
Scenario: Checking the edit functionality for my personal details  
   Given I launch the My Account URL
   And   I submit user credentials
   And   I click on Personal Details tab
   And   I successfully modify my Personal Details
   And   I am able to choose all the contact preferences available
   And   I click on Subscription Details tab
   When  I click on Personal Details tab
   Then  I can see that the chosen contact preferences remains stored

Radu-Stroe committed:
Fifth task commit. Waiting for review.

cornel18i commented on src/test/java/internship/framework/core/utility/TestData.java:
```diff
@@ -5,4 +5,11 @@
     }
 
+    //region yourDetails
+    public static final String YOUR_DETAILS_FIRST_NAME = "Radu";
```
Why did we create a dedicated class just to store some constants? Are they related just for one test?

Radu-Stroe replied:
TestData class was already part of Cristina's testing framework. She advised me to use that class for these constants.

cornel18i commented on src/test/java/internship/framework/pages/PersonalDetailsPage.java:
```diff
@@ -0,0 +1,191 @@
+    public PersonalDetailsPage selectCountry() {
+        ProjectLogger.info("Select country.");
+        Select select = new Select(countryElement);
+        select.selectByVisibleText("Romania");
```
Why is "Romania" not a constant ?

Radu-Stroe replied:
Noted and fixed.

cornel18i commented on src/test/java/internship/framework/pages/PersonalDetailsPage.java:
```diff
@@ -0,0 +1,191 @@
+
+    private boolean isFirstCheckBoxSelected() {
+        ProjectLogger.info("Check if first contact checkbox is selected.");
+        Driver.get().manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
```
Why do we do this only for the first checkbox ?

Radu-Stroe replied:
I need the implicit wait time between navigating to Personal details page and checking the first checkbox. I moved that line inside test method.

cornel18i commented on src/test/java/internship/framework/pages/PersonalDetailsPage.java:
```diff
@@ -0,0 +1,191 @@
+    }
+
+    public boolean areALLCheckboxesSelected() {
+        return isFirstCheckBoxSelected() & isSecondCheckBoxSelected()
```
Why did you use & instead of && ? Please read the following article : https://www.dummies.com/programming/java/and-operators-and-in-java/

Radu-Stroe replied:
My initial code include &&, but I got an advice from IntelliJ to simplify code this way. I've read the article and understand that is beneficial to use && performance-wise.

cornel18i commented on src/test/java/internship/tests/PersonalDetailsTest.java:
```diff
@@ -0,0 +1,60 @@
+    @Test(testName = "Personal details test",
+            description = "Testing personal details.")
+    public void checkPersonalDetailsFunctionality() {
+        softAssert.set(new SoftAssert());
```
Can we move line 17,18 from the test method somewhere else? This doesn't look related for the test.

Radu-Stroe replied:
Line 17 . Cristina's advice was to set a new SoftAssert object at the begining of every test and remove it after every testing, using tearDown method.
Line 18. login() method is part of this test requirement.

cornel18i replied:
**Line 17 . Cristina's advice was to set a new SoftAssert object at the begining of every test and remove it after every testing, using tearDown method.** So, how can we do this for line 17? Instead of copy/pasting the same line for each test?

cornel18i replied:
Line 18 can remain. If its part only for this test.

cornel18i commented on src/test/java/internship/tests/PersonalDetailsTest.java:
```diff
@@ -0,0 +1,60 @@
+        super.tearDown();
+    }
+
+    private void login()
```
Can we move this method in a test utility class? Most likely you are going to need it for other tests. Also, it will be good if you can allow to manually set the username and password value. This way you are not limited to only one case.

Radu-Stroe replied:
Noted. I've created a new class TestUtils and moved the login() method there. Changed the method to allow manually set the username and password.

cornel18i replied:
I think you forgot the last part.

Radu-Stroe replied:
Fixed.

cornel18i commented:
Also have a look at : https://app.codacy.com/app/cristina.nedin/internship2019/pullRequest?prid=3977707 . Codacy found 5 issues.

Radu-Stroe committed:
Create Title and Country constants.
Create TestUtils class and move method login() there. Change to manually set username and password.
Change Personal Details page to use title and country constants from above.
Move implicit wait from isFirstCheckBoxSelected() method to current test method.
Modify areaAllCheckboxesSelected() method, performance-wise.

cornel18i commented on src/test/java/internship/framework/core/utility/TestUtils.java:
```diff
@@ -0,0 +1,18 @@
+    public static void login() {
+        LoginPage loginPage = new LoginPage().openPage();
+        loginPage.clickLogInButtonHome();
+        loginPage.enterEmail("stroeradu03@gmail.com")
```
The method should have the option to set the email(username) and password. By doing so you are not constraint with a single case. e.g: test scenarios with different users.

Radu-Stroe replied:
Fixed.

Radu-Stroe committed:
Fixed changes.
